app/scripts/models/account.js

```diff
@@ -323,6 +325,10 @@ define(function (require, exports, module) {
       .then(function (updatedSessionData) {
         self.set(updatedSessionData);
 
+        if (options.reason === self._fxaClient.SIGNIN_REASON.ACCOUNT_UNLOCK) {
+          return updatedSessionData;
+        }
+
         if (! self.get('verified')) {
```

Comment:
Calling `signUpResend` was done before we had the signin-mixin, account.js provided a convenient place to place that functionality so that it was not duplicated. The problem to me is calling `signUpResend` always felt slightly out of place here, like concerns were mixed. Is this an opportune time to move `signUpResend` to the signin-mixin? Doing so would eliminate the need to check for the ACCOUNT_UNLOCK reason, and would reduce the number of side-effects caused.

Where all is account.signIn called?